v-next/hardhat-ethers/test-tmp/helpers/artifact-manager-mock.ts

```diff
@@ -0,0 +1,122 @@
+import type {
+  Artifact,
+  ArtifactsManager,
+  BuildInfo,
+  CompilerInput,
+  CompilerOutput,
+} from "@ignored/hardhat-vnext/types/artifacts";
+
+import { HardhatError } from "@ignored/hardhat-vnext-errors";
+
+export class MockArtifactsManager implements ArtifactsManager {
+  readonly #artifacts: Map<string, Artifact>;
+  readonly #artifactsPaths: Map<string, string>;
+
+  constructor(
+    mockedArtifacts?: Array<{ artifactName: string; fileName: string }>,
+  ) {
+    this.#artifacts = new Map();
+    this.#artifactsPaths = new Map();
+
+    // An array of elements, where every element has an artifact name and artifact file name, is passed as argument during initialization and stored in the map.
```

Comment:
This is the code I copied from the mocked ArtifactsManager PR. As mentioned in the PR description, I modified the logic slightly to suit my specific test scenarios